### PR TITLE
Add testing guideline for naming test files and adding comments

### PR DESCRIPTION
**Pull request by soareschen**
I can add more about other aspects of testing later on. These are the two draft test guidelines that I come out with while still at TPAC.

We might also want to consider whether to put the main guidelines in webrtc-pc or web-platform-tests. There may be other contributors who work on web-platform-tests but not webrtc-pc. So it might be better off to put the guidelines in WPT and link to it from webrtc-pc.

### CODE REVIEW

**Comment by alvestrand** [comment]
@fippo can you review this?
If you know of guidelines that cover parts of this that we should link to, please tell!

**Comment by fippo** [comment]
philip(p) confusion. Paging @foolip

**Comment by stefhak** [comment]
@foolip could you take a look?

**Comment by foolip** [comment]
It is really up to the people doing most of the work here to decide what amount of metadata and structure is useful. If I were to recommend anything, it would be to err on the side of less metadata to update, to optimize for ease of adding new tests.

Given the permission given to not bother with metadata in new tests or when updating tests, unless someone is continuously going to update the metadata, then it will get out of sync. As soon as that begins to happen, I think it might be better to delete the metadata.

**Comment by soareschen** [comment]
@foolip yes agree we can remove a number of metadata from the individual test files, in particular the todo list. I think some coverage accounting is still essential until we reach 100% coverage. Some considerations are how essential are such metadata to others, and where should we put the metadata. Do we put them in web-platform-tests (w3c/web-platform-tests#8051)? or as unofficial forks (https://rwaldron.github.io/webrtc-pc/)? or in webrtc-pc?

**Comment by foolip** [comment]
@soareschen, all the attempts to determine coverage from test-to-spec linkage so far don't seem very successful. Linking to just sections isn't granular enough. If you search for "data-tested-assertations" you can see such an attempt dating back to https://github.com/w3c/web-platform-tests/commit/6bb53056b2b9127bec0ddc5995d094421f0dc0a3. It's brittle approach, and I can only assume it's bitrotted.

The approach in https://github.com/tabatkins/bikeshed/issues/1116 is more promising to me, and should allow for views similar to https://rwaldron.github.io/webrtc-pc/. But that will be a Bikeshed feature, and WebRTC uses ReSpec. AFAICT, the least amount of work at this point would be to keep https://rwaldron.github.io/webrtc-pc/ up to date manually, by noting which wpt commit it's in sync with, and going through the changes since then for each update.

**Comment by soareschen** [comment]
Updating the coverage can be pretty tedious. If not done incrementally, it could take a whole week to track changes to new editor's draft. I guess that means the best approach is to pull out the metadata and put them in unofficial forks. If the fork gets bitrotten, it is simply forgotten and affect neither web-platform-tests nor webrtc-pc.

File: TESTING.md [hunk 1]
```diff
@@ -11,3 +11,40 @@ https://wpt.fyi/webrtc/RTCCertificate.html
...
+become widely available for WPT) Helper files should have as little dependency
+to other helper files, as script tags are not good for dependency management.
+
+## Comments
+
+The current tests have detailed comments referencing sections of the spec that
+is being tested. New tests may be written with minimal or no comments up to the
+author's preference. If a test case is testing specific steps in a spec, it is
+recommended to leave at least one line of comment referencing the spec to help
+readers quickly understand the objective of the test.
```
Comment (adam-be) [inline TESTING.md]: How does this relate to the "help link" that should point to the tested section in the spec?

Comment (adam-be) [reply TESTING.md]: Mention about "help link": https://github.com/w3c/web-platform-tests/pull/8450#discussion_r153404434

**Pull request merged**